### PR TITLE
`check_rmsThresh` use 2nd param; use peak

### PR DESCRIPTION
**Pull request by cwnaber**
Condense 2nd and 4th input params.

Get rid of the 'mean' method of calculating RMS value. Make a new method called 'peak_window'.

**This branch should not be merged immediately**. It should be timed with a change to current-studies which changes how input parameters are used for a small number of functions.

### CODE REVIEW

File: experiment_helpers/check_rmsThresh.m [hunk 1]
```diff
@@ -7,72 +7,62 @@
...
+%               during a period centered on the peak. If 'peak',
+%               the RMS value is the absolute peak RMS.
+%           * limits. A 2x2 array structured like this:
+%                  [GoodLow, GoodHi;
+%                   WarnLow, WarnHi]
+%              In check_rmsThresh, an area is shaded green between the low
+%              and hi Good limits, and an area is shaded yellow between Warn limits.
+%          * rmsThresh. If the RMS value is below rmsThresh, the output
+%              parameter bGoodTrial will be 0.
+%          * peakWindowSecs.
```
Comment (carrien) [inline experiment_helpers/check_rmsThresh.m:23]:
```suggestion
%          * peakWindowSecs. Width of the window centered on the peak, in seconds.
```

File: experiment_helpers/check_rmsThresh.m [hunk 2]
```diff
@@ -7,72 +7,62 @@
...
-%       * rmsThresh. If the RMS value is below rmsThresh, the output
-%           parameter bGoodTrial will be 0. This parameter is overridden by
-%           the 2nd input param `rmsThresh`, if that input param is used.
 %
-% 
 
-if nargin < 3, subAxis = []; end
-if nargin < 4
+if nargin < 2
+    params.checkMethod = 'peak';
```
Comment (carrien) [inline experiment_helpers/check_rmsThresh.m:31]: It looks like there's a different default for checkMethod depending on whether you a) don't provide a params argument (defaults to 'peak') or b) provide one with no checkMethod field (defaults to 'peak_window'). This seems confusing; why do we want this behavior?

File: experiment_helpers/check_rmsThresh.m [hunk 3]
```diff
@@ -7,72 +7,62 @@
...
 
-% for backwards compatibility, the second input parameter can override params.rmsThresh
-if nargin >= 2 && ~isempty(rmsThresh)
-    params.rmsThresh = rmsThresh;
-end
-
 %% find rmsValue
 switch params.checkMethod
     case 'peak'
         [rmsValue, onset] = max(data.rms(:,1));
```
Comment (carrien) [inline experiment_helpers/check_rmsThresh.m:49]:
```suggestion
        [rmsValue, peak] = max(data.rms(:,1));
        onset = peak;
```
Bringing this in line with (new) line 56.

File: experiment_helpers/check_rmsThresh.m [hunk 4]
```diff
@@ -7,72 +7,62 @@
...
-            end
-            rmsValue = mean(data.rms(onset:offset, 1));
-
-        % can't determine rms mean, because no ost-based vowel found, and RMS too low
-        else
-            rmsValue = NaN;
+    case 'peak_window'
+        % onset and offset are some number of ms before and after the peak.
+        % rmsValue is the mean RMS between onset and offset.
+        frameLenInSecs = data.params.frameLen/data.params.sRate;
```
Comment (carrien) [inline experiment_helpers/check_rmsThresh.m:54]: How is frameLen involved here?

**Review by carrien — CHANGES_REQUESTED** [review]

File: experiment_helpers/set_exptDefaults.m [hunk 5]
```diff
@@ -122,11 +122,15 @@
...
 
 %% amplitude tracking parameters
+%used with older experiments
 amplcalc.min_ampl = 0.04; %0.02;
 amplcalc.max_ampl = 0.24; %0.2;
 amplcalc.ons_thresh = 0.01;
 amplcalc.offs_thresh = 0.015;
+
+% used with newer experiments
 amplcalc.checkMethod = 'mean';   % Compare mean RMS during vowel against amplcalc.rmsThresh
```
Comment (carrien) [inline experiment_helpers/set_exptDefaults.m:132]:
```suggestion
amplcalc.checkMethod = 'peak_window';   % Compare mean RMS during vowel against amplcalc.rmsThresh
```

**Review by carrien — CHANGES_REQUESTED** [review]

File: experiment_helpers/check_rmsThresh.m (hunk 2 above)
Comment (cwnaber) [reply experiment_helpers/check_rmsThresh.m:31]: Will remove this to only have one default.

File: experiment_helpers/check_rmsThresh.m [hunk 6]
```diff
@@ -7,72 +7,63 @@
...
-if nargin < 4
+if nargin < 2
+    params = [];
+elseif isnumeric(params)
+    rmsThresh = params;
     params = struct;
-    defaultParams.checkMethod = 'peak';     % maintain legacy behavior
-else
-    defaultParams.checkMethod = 'mean';
+    params.checkMethod = 'peak';
```
Comment (carrien) [inline experiment_helpers/check_rmsThresh.m:35]:
```suggestion
    params.checkMethod = 'peak'; % maintain legacy behavior
```

File: experiment_helpers/set_exptDefaults.m [hunk 7]
```diff
@@ -122,11 +122,15 @@
...
+%used with older experiments
 amplcalc.min_ampl = 0.04; %0.02;
 amplcalc.max_ampl = 0.24; %0.2;
 amplcalc.ons_thresh = 0.01;
 amplcalc.offs_thresh = 0.015;
-amplcalc.checkMethod = 'mean';   % Compare mean RMS during vowel against amplcalc.rmsThresh
+
+% used with newer experiments
+amplcalc.checkMethod = 'peak_window';   % Compare mean RMS during vowel against amplcalc.rmsThresh
+amplcalc.peakWindowSecs = 0.100; % Number of seconds of buffer to include surrounding the peak amplitude to find RMS value
```
Comment (carrien) [inline experiment_helpers/set_exptDefaults.m:133]:
```suggestion
amplcalc.peakWindowSecs = 0.100; % Size of window (in seconds) surrounding the peak amplitude to find RMS value
```

**Review by carrien — DISMISSED** [review]

**Review by carrien — APPROVED** [review]

**Pull request merged**